### DISCUSSION TITLE
Only show MultiSelect placeholder when no values are selected

### DISCUSSION THREAD

**Discussion by devklick** [mantinedev/mantine#6542] [Q&A] [answered]
At present, when specifying a `placeholder` for a `MultiSelect` component, the placeholder will always show, regardless of whether or not any values have been selected. This is different to the `Select` component, where the placeholder is only shown if no value has been selected. Similarly, for a `TextInput` component, the placeholder is only shown when there is no value. 

I'm assuming it was designed this way to indicate to the user that more values can be selected, rather than it being a bug.

However, can we please introduce an option to suppress the placeholder value on a MultiSelect when values have been selected? My understanding of placeholders is that they should be temporary and only used when no value is present.

**Comment by rtivital** [comment]
You can remove placeholder by `placeholder={hasValues ? '' : 'Placeholder'}`.

**Reply by devklick** [reply]
Thanks @rtivital  for your reply. 

What is `hasValues` here - are you suggesting that I keep track of whether or not values have been selected in state outside of the component? E.g.
```jsx
const [hasValue, setHasValue] = useState(false);
const data = ["One", "Two", "Three"];
return (
  <MultiSelect
    data={data}
    placeholder={hasValue ? "" : "Pick some values"}
    onChange={(values) => setHasValue(values.length > 0)}
  />
);
```
Doing so is OK, but it does seem to involve an extra re-render whenever we go from 0 values selected to some values selected (and vice versa). Can this be avoided? It's also not ideal if I have multiple MultiSelect fields on a page - I have to keep track of whether each list has or does not have values selected, which could get a bit cumbersome.

I notice that the `Select` component doesnt suffer from re-renders when going from 0 values selected (or vice versa), yet it still clears the placeholder when a value has been selected. Is it possible to achieve the same thing for the `MultiSelect` component?

Note, to measure this I'm just logging on every render:
```jsx
const [hasValue, setHasValue] = useState(false);
const renders = useRef(1);

useEffect(() => {
  console.log(`Render count: ${renders.current++}`);
});

const data = ["One", "Two", "Three"];
return (
  <MultiSelect
    data={data}
    placeholder={hasValue ? "" : "Pick some values"}
    onChange={(values) => setHasValue(values.length > 0)}
  />
);

// return <Select data={data} placeholder="Pick a value" />;
```

**Comment by rtivital** [comment]
Here is a simple solution with CSS only – https://codesandbox.io/s/mantine-react-template-forked-3nzmwl?file=/src/App.tsx

**Reply by devklick** [reply]
I'm unable to access this sandbox - is it public?

**Reply by rtivital** [reply]
Sorry, I've made it public now, you can check.

**Comment by rtivital** [comment, marked as answer]
https://help.mantine.dev/q/multiselect-value-placeholder

**Reply by aleksanderd** [reply]
thank you! but changing color is not enough in some cases since placeholder still takes space
![image](https://github.com/user-attachments/assets/2774fc30-a429-458f-8ba1-ac0a2e364397)

**Reply by aleksanderd** [reply]
forced to override MultiSelect:
```tsx
import { FC } from 'react';
import { MultiSelect as MultiSelectBase, MultiSelectProps as BaseProps } from '@mantine/core';

export interface MultiSelectProps extends BaseProps {
  persistPlaceholder?: boolean;
}

export const MultiSelect: FC<MultiSelectProps> = ({
  value,
  placeholder: placeholderProp,
  persistPlaceholder,
  ...rest
}) => {
  const placeholder = !value?.length || persistPlaceholder ? placeholderProp : '';

  return (
    <MultiSelectBase
      placeholder={!value || !value.length || persistPlaceholder ? placeholder : ''}
      value={value}
      {...rest}
    />
  );
};

```
